File: src/test/java/quebec/virtualite/backend/services/domain/DomainServiceImplTest.java
package quebec.virtualite.backend.services.domain;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;
import quebec.virtualite.backend.services.domain.database.GreetingRepository;
import quebec.virtualite.backend.services.domain.entities.GreetingEntity;

import java.util.List;

import static org.hamcrest.core.IsNot.not;
import static org.hamcrest.core.IsNull.nullValue;
import static org.junit.Assert.assertThat;
import static org.mockito.Mockito.verify;

@RunWith(MockitoJUnitRunner.class)
public class DomainServiceImplTest
{
    private static final String NAME = "name";

    @InjectMocks
    private final DomainService domainService = new DomainServiceImpl();

    @Mock
    private GreetingRepository mockedGreetingRepository;

    @Test
    public void deleteGreetings()
    {
        // When
        domainService.deleteGreetings();

        // Then
        verify(mockedGreetingRepository).deleteAll();
    }

    @Test
    public void getGreetings()
    {
        // When
        List<GreetingEntity> greetings = domainService.getGreetings();

        // Then
        verify(mockedGreetingRepository).findAll();

        assertThat(greetings, not(nullValue()));
    }

    @Test
    public void recordGreeting()
    {
        // When
        domainService.recordGreeting(NAME);

        // Then
        verify(mockedGreetingRepository).findByName(NAME);
        verify(mockedGreetingRepository).save(new GreetingEntity(NAME));
    }
}

File: src/test/java/quebec/virtualite/backend/services/domain/DomainServiceImplTest.kt
package quebec.virtualite.backend.services.domain

import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.core.IsNot.not
import org.hamcrest.core.IsNull.nullValue
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.InjectMocks
import org.mockito.Mock
import org.mockito.Mockito.verify
import org.mockito.junit.MockitoJUnitRunner
import quebec.virtualite.backend.services.domain.database.GreetingRepository
import quebec.virtualite.backend.services.domain.entities.GreetingEntity

private const val NAME = "name"

@RunWith(MockitoJUnitRunner::class)
class DomainServiceImplTest
{
    @InjectMocks
    var domainService = DomainServiceImpl()

    @Mock
    lateinit var mockedGreetingRepository: GreetingRepository

    @Test
    fun deleteGreetings()
    {
        // When
        domainService.deleteGreetings()

        // Then
        verify(mockedGreetingRepository).deleteAll()
    }

    @Test
    fun getGreetings()
    {
        // When
        val greetings = domainService.getGreetings()

        // Then
        verify(mockedGreetingRepository).findAll()

        assertThat(greetings, not(nullValue()))
    }

    @Test
    fun recordGreeting()
    {
        // When
        domainService.recordGreeting(NAME)

        // Then
        verify(mockedGreetingRepository).findByName(NAME)
        verify(mockedGreetingRepository).save(GreetingEntity(NAME))
    }
}